### PR TITLE
chore(CI/CD): add semantic release automation

### PR DESCRIPTION
**Pull request by josecorella**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ajewellamz — APPROVED** [review]

File: .github/workflows/semantic_release.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        description: "Are you releasing a new version? (y/n)"
+        required: true
+        type: string
+
+jobs:
+  semantic-release:
+    # there is no easy way in gha to check if the actor is part of the team, running semantic release is a more
+    # privileged operation, so we must make sure this list of users is a subset of the users labeled as maintainers of
+    # https://github.com/orgs/aws/teams/aws-crypto-tools
+    if: contains('["seebees","texastony","ShubhamChaturvedi7","lucasmcdonald3","josecorella","imabhichow","rishav-karanjit","antonf-amzn","justplaz","ajewellamz"]', github.actor)
```
Comment (lucasmcdonald3) [inline .github/workflows/semantic_release.yml:17]:
```suggestion
    if: contains('["seebees","texastony","ShubhamChaturvedi7","lucasmcdonald3","josecorella","imabhichow","rishav-karanjit","antonf-amzn","justplaz","ajewellamz","RitvikKapila"]', github.actor)
```

File: .github/workflows/semantic_release.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# This workflow runs semantic release, bumps, generates changelog, and tags the project
+name: Semantic Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      dry-run:
+        description: "Are you releasing a new version? (y/n)"
```
Comment (lucasmcdonald3) [inline .github/workflows/semantic_release.yml:8]: Let's do either

```suggestion
      release-new-version:
        description: "Are you releasing a new version? (y/n)"
```

or

```suggestion
      dry-run:
        description: "Is this a dry run to validate semantic-release behaves as expected? (y/n)"
```

Right now, the Y/N are in opposition

File: .releaserc.cjs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  This file requires a `--branches` option to function.
+  This is to facilitate point releases if needed.
+
+  `npx semantic-release --branches main`
+*/
+
+// This project has several runtimes
+// each one has files that need to be updated.
+// We model all the files and the runtimes here in this structure
+const Runtimes = {
```
Comment (lucasmcdonald3) [inline .releaserc.cjs:21]:
```suggestion
// We model all the files and the runtimes here in this structure
// TODO: Add Java
const Runtimes = {
```

Comment (josecorella) [reply .releaserc.cjs:21]: good catch. It should not be a todo, I forgot to add java since it was copy paste from esdk

Comment (josecorella) [reply .releaserc.cjs:21]: fixed in latest commit

File: .releaserc.cjs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  java: {
+    "project.properties": {
```
Comment (lucasmcdonald3) [inline .releaserc.cjs:23]: From the [MPL](https://github.com/aws/aws-cryptographic-material-providers-library/blob/main/.releaserc.cjs), I would have expected this to be

```suggestion
    "DynamoDbEncryption/runtimes/java/build.gradle.kts": {
```

Why is DBESDK different?

File: .github/workflows/semantic_release.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        run: |
+          make setup_semantic_release
+
+      # Run semantic release in dry run mode if input matches
+      - name: Run Semantic Release in dry run mode
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        if: ${{inputs.dry-run == 'n'}}
+        run: |
+          make dry_run_semantic_release
```
Comment (lucasmcdonald3) [inline .github/workflows/semantic_release.yml:69]: This says if `dry-run` is `no`, then run a dry run
Similar flipping below (with worse consequences)

File: .releaserc.cjs (hunk 4 above)
Comment (josecorella) [reply .releaserc.cjs:23]: it is different since the java version is kept in the project.properties file

Comment (lucasmcdonald3) [reply .releaserc.cjs:23]: Got it. Thank you!

**Pull request merged**